src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -501,9 +501,16 @@ private static uint GetValEscape(TypeSymbol type, uint possibleValEscape)
         // required to identify it.  When that bug is fixed we should be able to remove this code and its callers.
         internal static bool IsZeroElementTupleType(TypeSymbol type)
         {
-            return type.IsStructType() && type.Name == "ValueTuple" && type.GetArity() == 0 &&
-                type.ContainingSymbol is var declContainer && declContainer.Kind == SymbolKind.Namespace && declContainer.Name == "System" &&
-                (declContainer.ContainingSymbol as NamespaceSymbol)?.IsGlobalNamespace == true;
+            return type.IsStructType() && type is
+            {
+                Name: "ValueTuple",
+                ContainingSymbol:
+                {
+                    ContainingSymbol: NamespaceSymbol { IsGlobalNamespace: true },
+                    Name: "System",
+                    Kind: SymbolKind.Namespace
+                } declContainer
+            } && type.GetArity() is 0;
```

Comment:
i feel like `is 0` is pretty non-idiomatic.  Feels like we should stick to well-trod patterns (no pun intended) unless something is subtantively better.  So, for example, if you're in a place where you *must* use a pattern, use a pattern.  But here we can just use an expression and we can keep it as `== 0`.

Reply:
This is a valid application of the refactoring, so the user has the option to make this decision.